Clarify protocols for CAMARA API access (#281)
camaraproject/IdentityAndConsentManagement: documentation/CAMARA-API-access-and-user-consent.md

Some CAMARA APIs process Personal Data and according to local regulations may require a “legal basis” to do so (e.g. "legitimate interest”, “contract”, “consent”). Operator API Exposure Platforms exposing CAMARA APIs should be built with a privacy-by-design approach to fully comply with any relevant data protection requirements and regulations, such as [GDPR](https://gdpr-info.eu/) in Europe.

CAMARA API access will be secured by the protocols described in the [CAMARA Security and Interoperability Profile](CAMARA-Security-Interoperability.md)

> - [OpenID Connect Authorization Code Flow](CAMARA-Security-Interoperability.md#oidc-authorization-code-flow) (OIDC) adds User authentication and consent collection on top of the [OAuth 2.0 protocol](https://datatracker.ietf.org/doc/html/rfc6749) in typical use cases where end-user's device sends an Authentication request to the Authorization Server starting User authentication and consent collection.

Reviewer: ```suggestion
- [OpenID Connect Authorization Code Flow](CAMARA-Security-Interoperability.md#oidc-authorization-code-flow) (OIDC) adds User authentication and Consent management on top of the [OAuth 2.0 protocol](https://datatracker.ietf.org/doc/html/rfc6749) in typical use cases where End-User's Device sends an Authentication request to the Authorization Server starting User authentication and Consent collection.
```

Author: I changed end-user to End-User and consent to Consent because they are a glossary terms

Author: Thanks for the review. The PR now just lists the protocols. Please review again.

> - [OpenId Connect Client-Initiated Backend Authentication](CAMARA-Security-Interoperability.md#client-initiated-backchannel-authentication-flow) (CIBA) adds User authentication and consent collection in typical cases where the API Consumer's backend sends an Authorization Request to the Authorization Server starting User authentication and consent collection.

Reviewer: ```suggestion
- [OpenId Connect Client-Initiated Backend Authentication](CAMARA-Security-Interoperability.md#client-initiated-backchannel-authentication-flow) (CIBA) adds User authentication and Consent management in typical cases where the API Consumer's backend sends an Authentication request to the Authorization Server starting User authentication and consent collection on an Authentication Device.
```

Author: Thanks for the review.

I simplified the list. It now just lists the protocols and does not mention "typical" use cases and scenarios.

Please review again

> - [OAuth2 Client-Credentials Flow](CAMARA-Security-Interoperability.md#client-credentials-flow) is typically used if the legal basis allows API usage without user consent. Use cases that have to consider that the end-user opted-out must check for opt-out when the access token created.

Reviewer: This is not consistent with the CAMARA agreed guideline, which is also included in this same document in the [`info.description` template](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#mandatory-template-for-infodescription-in-camara-api-specs):

> In cases where personal data is processed by the API and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of three-legged access tokens is mandatory. This ensures that the API remains in compliance with privacy regulations, upholding the principles of transparency and user-centric privacy-by-design.

If the user can opt-out, that guideline is saying that a 3-legged flow must be used. If you want to mention a typical use case for Client Credentials, I think it should be when no Personal Data is processed.

> - [OAuth2 Client-Credentials Flow](CAMARA-Security-Interoperability.md#client-credentials-flow) is typically used if the legal basis allows API usage without user consent. Use cases that have to consider that the end-user opted-out must check for opt-out when the access token created.

Reviewer: ```suggestion
- [OAuth2 Client-Credentials Flow](CAMARA-Security-Interoperability.md#client-credentials-flow) is typically used when no Personal Data is processed.
```

Reviewer 2: ```suggestion
- [OAuth2 Client-Credentials Flow](CAMARA-Security-Interoperability.md#client-credentials-flow) is typically used if the legal basis allows API usage without user consent. Use cases that have to consider that the end-user opted-out must check for opt-out when the access token created.
```
- [OAuth2 Client-Credentials Flow](CAMARA-Security-Interoperability.md#client-credentials-flow) is typically used when no Personal Data is processed or due to local regulations in the appliable jurisdiction, no check for consent or opt-out must be executed.

> The concept common to all flows is that the API access_token is created at the Authentication Server and that the API endpoint (Resource Server) grants access to the API based on the access_token alone.
> This separation of concern puts all the responsibility for implementing legal and business concerns under the authority of the Authentication Server and thus frees the Resource Server from the need to care about those. The API Provider's developers who implement the API can concentrate on the API which simplifies the service exposure platform implementation.

Reviewer: ```suggestion
The concept common to all flows is that the access token used to invoke an API is created at the Authorization Server, and the API endpoint (Resource Server) grants access to the API based solely on the access token.
This separation of concerns places all responsibility for implementing legal and business concerns under the authority of the Authorization Server, freeing the Resource Server from the need to worry about them. The API Provider's developers implementing the API can focus on the API functionality itself, simplifying the Resource Server's implementation.
```
